core/controllers/suggestion_test.py

```diff
@@ -320,6 +320,101 @@ def test_suggestion_list_handler(self):
             )['suggestions']
         self.assertEqual(len(suggestions), 2)
 
+    def test_invalid_action_type_of_suggestion(self):
+        self.login(self.EDITOR_EMAIL)
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        suggestion_to_accept = self.get_json(
+            '%s?author_id=%s' % (
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                self.author_id))['suggestions'][0]
+
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.put_json('%s/exploration/%s/%s' % (
+            feconf.SUGGESTION_ACTION_URL_PREFIX,
+            suggestion_to_accept['target_id'],
+            suggestion_to_accept['suggestion_id']), {
+                'action': u'invalid',
+                'commit_message': u'commit message',
+                'review_message': u'Accepted'
+            }, csrf_token=csrf_token, expected_status_int=400)
+
+    def test_accept_suggestion_on_invalid_exploration_handler(self):
+        suggestion_to_accept = self.get_json(
+            '%s?author_id=%s' % (
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                self.author_id))['suggestions'][0]
+        suggestion_id = suggestion_to_accept['suggestion_id']
+        invalid_suggestion_id = ('invalid.' +
+                                 '.'.join(suggestion_id.split('.')[1:]))
+
+        self.login(self.EDITOR_EMAIL)
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        self.put_json('%s/exploration/%s/%s' % (
+            feconf.SUGGESTION_ACTION_URL_PREFIX,
+            suggestion_to_accept['target_id'],
+            invalid_suggestion_id), {
+                'action': u'accept',
+                'commit_message': u'commit message',
+                'review_message': u'Accepted'
+            }, csrf_token=csrf_token, expected_status_int=500)
+
+    def test_reject_suggestion(self):
+        self.login(self.EDITOR_EMAIL)
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        suggestion_to_reject = self.get_json(
+            '%s?author_id=%s' % (
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                self.author_id))['suggestions'][0]
+
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.put_json('%s/exploration/%s/%s' % (
```

Comment:
Please check the status of the suggestion after this PUT. It should be STATUS_REJECTED